ember-toucan-core/src/components/form/input-field.ts

```diff
@@ -0,0 +1,39 @@
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+interface ToucanFormInputFieldComponentSignature {
+  Element: HTMLInputElement;
+  Args: {
+    error?: string;
+    label: string;
+    hint?: string;
+    isDisabled?: boolean;
+    onChange?: (value: string, e: Event | InputEvent) => void;
+    readonly?: boolean;
+    rootTestSelector?: string;
+    value?: string;
+  };
+  Blocks: {
+    default: [];
+  };
+}
+
+export default class ToucanFormInputFieldComponent extends Component<ToucanFormInputFieldComponentSignature> {
+  constructor(
+    owner: unknown,
+    args: ToucanFormInputFieldComponentSignature['Args']
+  ) {
+    assert('input field requires a label', args.label !== undefined);
+    super(owner, args);
+  }
+
+  @action
+  handleInput(e: Event | InputEvent): void {
+    assert('Expected HTMLInputElement', e.target instanceof HTMLInputElement);
+
+    if (this.args.onChange) {
+      this.args.onChange(e.target.value, e);
+    }
+  }
```

Comment:
Do we still need this here since we can pass the argument directly through to the underlying Input component?

Reply:
@ynotdraw remove the `handleInput` (or the entire backing class) you mean?

Reply:
oh wait I see...going to remove the backing class.